### DISCUSSION TITLE
Parts cut & assembly hole/tenon ?

### DISCUSSION THREAD

**Discussion by DzzD** [supermerill/SuperSlicer#1812] [Q&A]
I have seen that it is possible to cut a model in two parts the up & down but did not find a way to add an assembly part like a hole in part A and a corresponding tenon onto B part, is that possible with SuperSlicer ? If not possible it would be a really helpfull feature no ?

**Comment by supermerill** [comment]
I've made a wiki entry for that, maybe that's where you see it?
https://github.com/supermerill/SuperSlicer/wiki/Splitting-an-object-with-pins-&-holes
It needs you to have freecad installed, as it's used to modify the objects.

> did not find a way to add an assembly part like a hole in part A and a corresponding tenon onto B part

In the script, as it takes the same object, it can add and/or remove cylinder (or cubes if you prefer) from it, at the exact same place.

**Reply by DzzD** [reply]
Hum.... There must be a problem with my notifications as I did not see your answers...,

Nop I did not see it on the wiki, I just thought of it first time I used slicing command in Super Slicer.

I ll read the wiki carefully, I ahve already installed freecad, but it would be really awsome to be able to make boolean operation with simple shapes : cube, sphere, cylinder directly within the 3D view GUI of SuperSlicer

There are tools that easily can do that but having it right in place in SuperSlicer would be great.

**Reply by supermerill** [reply]
>  it would be really awsome to be able to make boolean operation

That would mean integrating opencascade to transform the slicer into a cad software.

**Reply by DzzD** [reply]
Yes that's a bit too much :) :) ok ok, but at least based on the script you have already done a button or an option to add holes when slicing on a z layer + separated pins would be enought

**Reply by supermerill** [reply]
We can create a gui widow where we can enter the size & length of the tenon, the height of the cut, etc...  and then ask for generation.

**Comment by DzzD** [comment]
After thinking a bit more about this feature idea, I think we should create holes in both parts and create pins separatly.

This will be a lot better because user will be able to print flat both part (and pins) without support and assembly will be as easy.

no ?

ps: do I need to change this discussion to an issue ?

**Reply by supermerill** [reply]
For double-hole, don't use the union but put eveeyrhting in the cut in the first part, with the same "fat" cylinder.
And add slim cylinders at the bottom, to be added to the plater.
